File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/lexer/DebugLexerAdapter.java
package net.sjrx.intellij.plugins.systemdunitfiles.lexer;

import com.intellij.lexer.LexerBase;
import com.intellij.openapi.diagnostic.Logger;
import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class DebugLexerAdapter extends LexerBase {

  private static final Logger LOG = Logger.getInstance(DebugLexerAdapter.class);
  private final UnitFileLexerAdapter ufla = new UnitFileLexerAdapter();

  @Override
  public void start(@NotNull CharSequence buffer, int startOffset, int endOffset, int initialState) {
    LOG.info("start(" + buffer + "," + startOffset + "," + endOffset + "," + initialState  + ") ==> void");
    ufla.start(buffer, startOffset, endOffset, initialState);
  }

  @Override
  public int getState() {

    int state =  ufla.getState();

    LOG.info("getState() ==> " + state);

    return state;
  }

  @Nullable
  @Override
  public IElementType getTokenType() {

    IElementType ooType =  ufla.getTokenType();

    LOG.info("getTokenType() ==> " + ooType);

    return ooType;
  }

  @Override
  public int getTokenStart() {
    int tokenStart = ufla.getTokenStart();

    LOG.info("getTokenStart() ==> " + tokenStart);

    return tokenStart;
  }

  @Override
  public int getTokenEnd() {
    int tokenEnd =  ufla.getTokenEnd();

    LOG.info("getTokenEnd() ==> " + tokenEnd);

    return tokenEnd;
  }

  @Override
  public void advance() {
    ufla.advance();

    LOG.info("advance() ==> void");
  }

  @NotNull
  @Override
  public CharSequence getBufferSequence() {
    CharSequence cs = ufla.getBufferSequence();

    LOG.info("getBufferSequence() ==> " + cs);

    return cs;
  }

  @Override
  public int getBufferEnd() {

    int be = ufla.getBufferEnd();

    LOG.info("getBufferEnd() ==> " + be);

    return be;
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/lexer/DebugLexerAdapter.kt
package net.sjrx.intellij.plugins.systemdunitfiles.lexer

import com.intellij.lexer.LexerBase
import com.intellij.openapi.diagnostic.Logger
import com.intellij.psi.tree.IElementType

class DebugLexerAdapter : LexerBase() {
  private val ufla = UnitFileLexerAdapter()
  override fun start(buffer: CharSequence, startOffset: Int, endOffset: Int, initialState: Int) {
    LOG.info("start($buffer,$startOffset,$endOffset,$initialState) ==> void")
    ufla.start(buffer, startOffset, endOffset, initialState)
  }

  override fun getState(): Int {
    val state = ufla.state
    LOG.info("getState() ==> $state")
    return state
  }

  override fun getTokenType(): IElementType? {
    val ooType = ufla.tokenType
    LOG.info("getTokenType() ==> $ooType")
    return ooType
  }

  override fun getTokenStart(): Int {
    val tokenStart = ufla.tokenStart
    LOG.info("getTokenStart() ==> $tokenStart")
    return tokenStart
  }

  override fun getTokenEnd(): Int {
    val tokenEnd = ufla.tokenEnd
    LOG.info("getTokenEnd() ==> $tokenEnd")
    return tokenEnd
  }

  override fun advance() {
    ufla.advance()
    LOG.info("advance() ==> void")
  }

  override fun getBufferSequence(): CharSequence {
    val cs = ufla.bufferSequence
    LOG.info("getBufferSequence() ==> $cs")
    return cs
  }

  override fun getBufferEnd(): Int {
    val be = ufla.bufferEnd
    LOG.info("getBufferEnd() ==> $be")
    return be
  }

  companion object {
    private val LOG = Logger.getInstance(DebugLexerAdapter::class.java)
  }
}
